cmd/snap-update-ns/bootstrap.c

```diff
@@ -368,6 +368,40 @@ void process_arguments(int argc, char *const *argv, const char **snap_name_out,
 				// Processing the user-fstab file implies we're being
 				// called from snap-confine.
 				should_setns = false;
+			} else if (!strcmp(arg, "-u")) {
+				const char *uid_text = argv[i + 1];
```

Comment:
I think wee should check `if i+1==argc` before referencing argv[i+1], not below.

Reply:
The argument vector is always terminated with a NULL pointer so there will always be one extra slot to look at but I agree that for extra sanity it should be done after the check.